bentoml/deployment/aws_lambda/utils.py

```diff
@@ -87,12 +87,18 @@ def cleanup_build_files(project_dir, api_name):
 
 def call_sam_command(command, project_dir):
     command = ['sam'] + command
-    with subprocess.Popen(
+    proc = subprocess.Popen(
         command, cwd=project_dir, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-    ) as proc:
-        stdout = proc.stdout.read().decode('utf-8')
-        logger.debug('SAM cmd %s output: %s', command, stdout)
-    return stdout
+    )
+    stdout, stderr = proc.communicate()
+    logger.debug('SAM cmd %s output: %s', command, stdout.decode('utf-8'))
+    if proc.returncode == 1:
+        error_message = stderr.decode('utf-8')
+        if not error_message:
+            error_message = stdout.decode('utf-8')
+        raise BentoMLException(error_message)
```

Comment:
could we make this function simply return (error_code, stdout, stderror) and have the caller to decide what to do next?